src/coredump/coredump.c

```diff
@@ -1054,17 +1054,15 @@ static char* set_iovec_field_free(struct iovec iovec[27], size_t *n_iovec, const
 }
 
 static int gather_pid_metadata_and_process_special_crash(
-                const char *context[_CONTEXT_MAX],
+                char* context[_CONTEXT_MAX],
                 char **comm_fallback,
-                char **comm_ret,
```

Comment:
can't say I like having a single array with such strange memory allocation rules for different fields... But maybe it's OK, but I think a big fat comment here might be a good choice, saying that most fields are just input, but 2 are just output...